snapcraft.yaml

```diff
@@ -20,9 +20,15 @@ parts:
     # and prime anyway.
     override-stage: |
       unset LD_LIBRARY_PATH;
-      export PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin";
+      export PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin"
+      # ensure snapcraftctl is found, see
+      # https://github.com/snapcore/snapcraft/pull/2251
+      export PATH="$PATH:/snap/snapcraft/current/bin/scriptlet-bin"
```

Comment:
The path belongs to a private binary from snapcraft, is this really what we want to do?

Reply:
I can expand the command a bit maybe - so what happens is that snapcraft 3.0 will run the scriptlets in a subshell with custom PATH and LD_LIBRARY_PATH. Unfortunately this breaks on non-18.04 systems because of incompatible binaries. Hence the workaround to rewrite PATH was added. However in snapcraft 3.0 snapcraftctl is put into a custom PATH (it used to be a shell alias). This is why this is needed, does that answer the question?